src/raft.rs

```diff
@@ -546,7 +538,15 @@ impl<T: Storage> Raft<T> {
             return;
         }
         let mut m = Message::default();
+        if self.leader_id != self.id {
+            // Self is a delegate, bring the leader id to the other member
+            // TODO: This should be safe as the delegate just calls `handle_append_message` which
+            // means it calls `become_follower`. Is that true?
```

Comment:
It's true because after a candidate peer receives MsgAppend from leader, it calls `become_follower`.